File: Queue.cpp
// Queue

#include<iostream>
using namespace std;

typedef struct node
{
    int data;
    struct node *next;

}NODE , * PNODE;

class Queue   //Insertlast
{
    public:
        PNODE head;
        int iCount;

        Queue()
        {
            head = NULL;
            iCount = 0;
        }

        void Enqueue(int iNo)      
        { 
            PNODE newn = NULL;
            PNODE temp=NULL;

            newn = new NODE;

            newn->data = iNo;
            newn->next = NULL;

            if(head == NULL)
            {
                head = newn;
            }
            else
            {
               temp=head;
               while(temp->next !=NULL)
               {
                temp=temp->next;
               }
               temp->next=newn;

            }
            iCount++;
        }

        int Dequeue()  //delete first
        { 
            PNODE temp = NULL;

            int no=-1;

            if(head == NULL)
            {
                cout<<"Stack is Empty\n";
                return -1;
            }
            else if(head->next == NULL)
            {
                no = head->data;
                delete head;
                head=NULL;
            }
            else
            {
                no=head->data;
                temp = head;
                head = head->next;
                delete temp;
            }
            iCount--;
            return no;
        }

        void Display()
        { 
            
            PNODE temp=head; 

            while(temp !=NULL)
            {
                cout<<temp->data<<"->";
                temp=temp->next;
            }
            cout<<"NULL"<<endl;
        }

        int Count()
        {
            return iCount;
        }
};

int main()
{
   
    int iret=0;

    Queue sobj;
    sobj.Enqueue(101);
    sobj.Enqueue(51);
    sobj.Enqueue(21);
    sobj.Enqueue(11);
   
    cout<<"Queue elements are \n";
    sobj.Display();
    iret=sobj.Count();
    
    cout<<"total elements "<<iret<<"\n";

    iret=sobj.Dequeue();
    cout<<"Dequeue elements is :"<<iret<<"\n";
    iret=sobj.Dequeue();
    cout<<"Dequeue elements is :"<<iret<<"\n";

    cout<<"Queue elements are \n";
    sobj.Display();

    iret=sobj.Count();
    cout<<"total elements "<<iret<<"\n";
    

    return 0;

}

File: Queue.java
class node
{
    public int data;
    public node next;
    
    public node(int X)
    {
        data=X;
        next=null;
    }
}

class QueueX
{

    public node head;
    public int icount;

    public QueueX()
    {
        head=null;
       icount=0;
    }

    public void Enqueue(int ino)
    {
         node newn=null;
         node temp=head;

         newn =new node(ino);

         if(head==null)
         {
            head=newn;
         }

         else
         {
            while(temp.next!=null)
            {
                temp=temp.next;
            }
            temp.next=newn;
         }
         icount++;
    }

    public void Dequeue()
    {
        if(head==null)
         {
           System.out.println("Queue is empty");
            return ;
         }

         else if(head.next==null)
         {
            head=null;
            System.gc();
         }

         else 
         {
            node temp=head;
            head=head.next;
            System.gc();
         }
        
        icount--;

    }
    public void Display()
    {
        node temp=head;

        while(temp!=null)
        {
            System.out.println(temp.data);
            temp=temp.next; 
        }
    }


    public int Count()
    {
        return icount;
    }

}
public class Mrunal 
{
    public static void main(String Arg[])
    {
         QueueX qobj=new QueueX();

         qobj.Enqueue(51);
        qobj.Enqueue(21);
        qobj.Enqueue(11);
        qobj.Dequeue();
        qobj.Display();
        int iret=0;
        iret=qobj.Count();

        System.out.println("Elements after pushing are :"+iret);
    }
    
}
